lib/events.js

```diff
@@ -103,30 +103,29 @@ EventEmitter.prototype.getMaxListeners = function getMaxListeners() {
   return $getMaxListeners(this);
 };
 
-// Returns the longest sequence of `a` that fully appears in `b`,
-// of length at least 3.
-// This is a lazy approach but should work well enough, given that stack
-// frames are usually unequal or otherwise appear in groups, and that
-// we only run this code in case of an unhandled exception.
-function longestSeqContainedIn(a, b) {
-  for (var len = a.length; len >= 3; --len) {
-    for (var i = 0; i < a.length - len; ++i) {
-      // Attempt to find a[i:i+len] in b
-      for (var j = 0; j < b.length - len; ++j) {
-        let matches = true;
-        for (var k = 0; k < len; ++k) {
-          if (a[i + k] !== b[j + k]) {
-            matches = false;
-            break;
-          }
+// Returns the length and line number of the first sequence of `a` that fully
+// appears in `b` with a length of at least 4.
+function identicalSequenceRange(a, b) {
+  for (var i = 0; i < a.length - 3; i++) {
+    // Find the first occurrence of the frame that could match up to four
+    // frames.
+    const pos = b.indexOf(a[i]);
```

Comment:
If you use a slice of `a`, e.g. `a.substr(i, i + 3)` instead of a only `a[i]`, you could get rid of the `if (len > 3)` check later?

Reply:
Sorry, I am a bit lost. Do you mean I should change the input from arrays to strings, read the line breaks and then work with substrings? I guess that might be a bit faster.

Reply:
@BridgeAR No, nothing like that :smile: I meant this applying to this specific line, literally replacing `a[i]` with `a.substr(i, i + 3)` here and dropping the `if (len > 3)` check later because that becomes unnecessary.

Reply:
How is that possible? `a` is an array, not a string.

Reply:
Oh, right, I see – yes, sorry, that’s my bad :smile: